edgePy/io/_io.py

```diff
@@ -7,15 +7,60 @@
 
 
 __all__ = [
-    'Importer',
-    'get_dataset_path'
+    'GroupImporter',
+    'DataImporter',
+    'get_dataset_path',
+    'get_open_function'
 ]
 
 
-class Importer(object):
+def get_open_function(filename):
+    decode_required = False
+    if filename.endswith("gz"):
+        open_function = gzip.open
+        decode_required = True
+    else:
+        open_function = open
+
+    return open_function, decode_required
+
+
+class GroupImporter(object):
+
+    def __init__(self, filename=None):
+        self.filename = str(filename) if filename else None
+        if not self.filename:
+            raise Exception("No group filename")
+        self.samples = {}
+        self.groups = {}
+        self.read_file()
+
+    def read_file(self):
+        open_function, decode_required = get_open_function(filename=self.filename)
+        with open_function(self.filename) as f:
+            for line in f:
+                if decode_required:
+                    # this is needed if we are using gzip, which returns a
+                    # binary-string.
+                    line = line.decode('utf-8')
+                line = line.strip().split(":")
+                if len(line) < 2:
+                    continue
+                group = line[0].strip()
+                samples = [x.strip() for x in line[1].split(",")]
+                self.groups[group] = samples
+                for sample in samples:
+                    if sample in self.samples:
+                        raise Exception(f"Duplicate sample detected! {sample}")
+                    self.samples[sample] = group
+
+
+class DataImporter(object):
```

Comment:
Without a docstring it's unclear what `Data` this is meant to import. Consider adding a docstring and possibly being more specific with the class name.

If it's meant to be a generic parser how about `DelimitedDataParser`?

If it's meant to be specific to a common file format, a definition of that file format is needed.

Also, if we intend to rely on NumPy later you can use this function which does most of what you have coded already:

https://docs.scipy.org/doc/numpy-1.13.0/user/basics.io.genfromtxt.html

Reply:
I fail to see the point of using the numpy version, unless we want to work entirely in numpy objects.

Reply:
Shifting the burden of code testing and code correctness would be one argument for the dependency, albeit a weak one for such a heavy add. 

I don't see how we'll get by on this project without fast vectorized math operations, and numpy was designed for that. Up to the masses though! 

If not, you could simplify with just the built-in `csv` module which is good at delimited data parsing.

Reply:
I think we need a target algorithm in order to make a compelling argument for a particular parser.  I wrote this minimal one in order to get things started.  Once we decide on formats and algorithms, this should be completely replaced anyhow - with the correct parser.